src/js/common/schemaform/widgets/ArrayCountWidget.jsx

```diff
@@ -1,45 +1,45 @@
 import React from 'react';
 
-// We're expanding or contracting the array based on the count
-// and returning undefined if the array should be empty
-function getValue(count, value = []) {
-  if (!count || parseInt(count, 10) < 1) {
-    return undefined;
+export default class SpouseMarriageWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: !!props.value
+      ? props.value.length - (this.props.options.countOffset || 0)
```

Comment:
After reading this through for _waaaay_ too long, I think I finally figured out why we're _subtracting_ the offset. Correct me if I'm wrong:

We start with (usually) a length of 0, so when we have an offset of -1 (as in we know the spouse has 1 marriage more than we'll ask information for), then the value we'll put into the `input` field will be `0 - (-1) = 1`. Later, when we update the field to trigger `getValue()`, that will call `this.props.onChange` with an array sized `this.state.value + this.props.offsetCount` so if we enter 3, the array of marriages we'll ask information for will have a length of `3 + (-1) = 2`.

Also, can we rename `value` to `arrayLength` or something more descriptive?